bottlecap/src/lifecycle/invocation/span_inferrer.rs

```diff
@@ -52,6 +54,7 @@ impl SpanInferrer {
             wrapped_inferred_span: None,
             is_async_span: false,
             carrier: None,
+            aws_header_context: None,
```

Comment:
Instead of creating a new field, can't you you use the same logic StepFunctions is doing?

Reply:
ok I will reuse the same var

Reply:
I had to add a boolean since step functions are ignoring inferred spans.
I would honestly prefer the 2 span approach since it's more explicit and easier to understand.
But I will go with the boolean approach for now